src/create-element.js

```diff
@@ -100,3 +79,35 @@ export function Fragment(props) {
  */
 export const isValidElement = vnode =>
 	vnode != null && vnode.constructor === undefined;
+
+export function createBackingNode(vnode, original) {
+	const backingNode = {
+		key: vnode.props ? vnode.props.key : undefined,
+		ref: vnode.props ? vnode.props.ref : undefined,
+		type: vnode.type,
```

Comment:
should we duplicate `type` on here? My gut sais yes for easy access

Reply:
Yep, avoids reaching into the vnode. I do think there's merit in adding a property (bitfield?) that indicates the vnode's type grouping (text, element, function, class) since that's a decent percentage of the current usage of `vnode.type`.

Reply:
@developit Should we aim to touch `vnode` as little as possible in general with the backing tree approach? E.g. copying relevant stuff onto the backing node to minimize `vnode` access?